libs/dal/src/repositories/integration/integration.repository.ts

```diff
@@ -28,4 +29,35 @@ export class IntegrationRepository extends BaseRepository<IntegrationEntity> {
       'Duplicate key - One application may not have two providers of the same channel type'
     );
   }
+
+  async update(
+    query: FilterQuery<IntegrationEntity & Document>,
+    updateBody: any
+  ): Promise<{
+    matched: number;
+    modified: number;
+  }> {
+    if (query.active === true) {
+      await this.deactivatedOtherActiveChannels(query);
+    }
+
+    return await super.update({ _id: query._id }, updateBody);
+  }
+
+  async deactivatedOtherActiveChannels(query: FilterQuery<IntegrationEntity & Document>): Promise<void> {
+    const otherExistedIntegration = (
+      await this.find({
+        _applicationId: query._applicationId,
+        channel: query.channel,
+        active: true,
+      })
+    ).filter((x) => x.providerId !== query.providerId);
+
+    for (const x of otherExistedIntegration) {
+      const deactivatedIntegration = x;
+
+      deactivatedIntegration.active = false;
+      await super.update({ _id: deactivatedIntegration._id }, deactivatedIntegration);
+    }
```

Comment:
```suggestion
    if (otherExistedIntegration.length) {
         super.update({ _id: {$in: otherExistedIntegration.map(i => i._id)}}, { $set: { active: false }});
    }
```

I believe this can be simplified to this ^ and perform it in a single DB query instead of n queries